src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ReferralForDashboard.kt

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity
+
+import org.hibernate.annotations.Fetch
+import org.hibernate.annotations.FetchMode
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.CascadeType
+import javax.persistence.CollectionTable
+import javax.persistence.ElementCollection
+import javax.persistence.Entity
+import javax.persistence.FetchType
+import javax.persistence.Id
+import javax.persistence.Index
+import javax.persistence.ManyToOne
+import javax.persistence.NamedAttributeNode
+import javax.persistence.NamedEntityGraph
+import javax.persistence.NamedSubgraph
+import javax.persistence.OneToOne
+import javax.persistence.PrimaryKeyJoinColumn
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@NamedEntityGraph(
+  name = "entity-referral-graph",
+  attributeNodes =
+  [
+    NamedAttributeNode("sentBy"),
+    NamedAttributeNode(value = "serviceUserData", subgraph = "serviceUserData"),
+    NamedAttributeNode(value = "intervention", subgraph = "interventions"),
+    NamedAttributeNode(value = "endOfServiceReport", subgraph = "endOfServiceReport")
+  ],
+  subgraphs = [
+    NamedSubgraph(
+      name = "interventions",
+      attributeNodes = [
+        NamedAttributeNode("dynamicFrameworkContract"),
+      ]
+    ),
+    NamedSubgraph(
+      name = "endOfServiceReport",
+      attributeNodes = [
+        NamedAttributeNode("referral"),
+        NamedAttributeNode("createdBy"),
+        NamedAttributeNode("submittedBy")
+      ]
+    ),
+    NamedSubgraph(
+      name = "serviceUserData",
+      attributeNodes = [
+        NamedAttributeNode("disabilities"),
+        NamedAttributeNode("referral"),
+      ]
+    )
+  ]
+)
+@Entity(name = "referral")
+@Table(name = "referral", indexes = arrayOf(Index(columnList = "created_by_id")))
+class ReferralForDashboard(
```

Comment:
nit on the naming of this entity: this service doesn't have the concept of dashboards, so i think we should keep the name generic

Reply:
I agree Tom but naming a code is one of the complex jobs for a developer 😢 , Can I name it as ReferralView?